src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioSymbolNavigationService.cs

```diff
@@ -82,6 +84,31 @@ public bool TryNavigateToSymbol(ISymbol symbol, Project project, CancellationTok
                 return false;
             }
 
+            // Should we prefer navigating to the Object Browser over metadata-as-source?
+            if (options.GetOption(VisualStudioNavigationOptions.NavigateToObjectBrowser, project.Language))
+            {
+                var libraryService = project.LanguageServices.GetService<ILibraryService>();
+                if (libraryService == null)
+                {
+                    return false;
+                }
+
+                var compilation = project.GetCompilationAsync(cancellationToken).WaitAndGetResult(cancellationToken);
+                var navInfo = libraryService.NavInfo.CreateForSymbol(symbol, project, compilation);
+                if (navInfo == null)
+                {
+                    navInfo = libraryService.NavInfo.CreateForProject(project);
+                }
+
+                if (navInfo == null)
+                {
+                    return false;
```

Comment:
If we didn't add a fallback, what does the user get? If it should never happen, we should at least crash to be sure. :smile: